queries/idl/highlights.scm

```diff
@@ -0,0 +1,206 @@
+; https://neovim.io/doc/user/treesitter.html#treesitter-highlight-groups
+(comment) @comment
+
+[
+  "enum"
+  "struct"
+  "union"
+  "bitmask"
+  "bitset"
+  "@annotation"
+  "interface"
+  "sequence"
+  "exception"
+] @keyword.type
+
+[
+  "module"
+  "switch"
+  "case"
```

Comment:
Case and switch should usually get `@keyword.conditional`

Reply:
as you see, switch and case is a part of union declare, I'm not sure whether it should be `@keyword.conditional` or not.

```idl
union A switch(long) {
    case 1:
        u8 a;
    case 2:
    case 3:
        u16 b;
    default:
        u32 c[10];
};
```

Reply:
I personally think they should still, but I don't have strong feelings either way. Also I think that `char_literal` and `string_literal` should be broken up, getting `@character` and `@string` highlights, respectively

Reply:
This looks similar enough to C-switch/case, so it should follow what C's switch-case captures

Reply:
The color looks a bit strange:
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/42114817/d3c00497-2b95-461e-ab14-38050c5058de)
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/42114817/f174e390-9068-4656-9b28-82e67ba7c3e5)

Reply:
We don't care about the color here :) Only the correctness of the capture

Reply:
ok

Reply:
Also maybe it would help if `default` was also `@keyword.conditional`? Apparently C does not do this but many other languages I have seen here do, and maybe C should be doing this as well...

Reply:
sure, I mark the default also be @keyworkd.conditional